daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/generators/CodeGeneratorState.scala

```diff
@@ -495,29 +509,27 @@ class CodeGeneratorState(private val root: ElementBase) {
          | PrimType.Int
          | PrimType.Long
          | PrimType.Integer =>
-        getLengthInBits(e) match {
-          case 8 =>  PrimType.Byte
-          case 16 => PrimType.Short
-          case 32 => PrimType.Int
-          case 64 => PrimType.Long
-          case _ =>  e.SDE("Integer lengths other than 8, 16, 32, or 64 bits are not supported.")
+        lengthInBits match {
+          case n if n <= 8 =>  PrimType.Byte
+          case n if n <= 16 => PrimType.Short
+          case n if n <= 32 => PrimType.Int
+          case n if n <= 64 => PrimType.Long
+          case _ =>  e.SDE("Integer lengths longer than 64 bits are not supported.")
         }
       case PrimType.UnsignedByte
          | PrimType.UnsignedShort
          | PrimType.UnsignedInt
          | PrimType.UnsignedLong
          | PrimType.NonNegativeInteger =>
-        getLengthInBits(e) match {
-          case 8 =>  PrimType.UnsignedByte
-          case 16 => PrimType.UnsignedShort
-          case 32 => PrimType.UnsignedInt
-          case 64 => PrimType.UnsignedLong
-          case _ =>  e.SDE("Unsigned integer lengths other than 8, 16, 32, or 64 bits are not supported.")
+        lengthInBits match {
+          case n if n <= 8 =>  PrimType.UnsignedByte
+          case n if n <= 16 => PrimType.UnsignedShort
+          case n if n <= 32 => PrimType.UnsignedInt
+          case n if n <= 64 => PrimType.UnsignedLong
+          case _ =>  e.SDE("Unsigned integer lengths longer than 64 bits are not supported.")
         }
       case p => e.SDE("PrimType %s is not supported in C code generator.", p.toString)
     }
-    if (primType != e.optPrimType.get)
-      e.SDW(WarnID.IgnoreDFDLProperty, "Ignoring PrimType %s, using %s", e.optPrimType.get.toString, primType.toString)
```

Comment:
Why remove this warning? This seems like an mistake in a schema that a user should be notified of?